server/form.go

```diff
@@ -0,0 +1,73 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// IndexForm represents a request form where data are retrieved through indexes.
+type IndexForm interface {
+	// GetArg returns an argument giving the index.
+	GetArg(index int) Argument
+}
+
+// MapForm represents a request form where data are retrieved through field names.
+type MapForm interface {
+	// GetArg returns an argument giving the index.
+	GetArg(field string) Argument
+}
+
+type indexForm struct {
+	args []interface{}
+}
+
+type mapForm struct {
+	fields map[string]interface{}
+}
+
+// NewIndexForm creates a new form to retrieve values from its index.
+func NewIndexForm(contentType string, body []byte) (IndexForm, error) {
+	if contentType != "application/porthos-args" {
```

Comment:
yeap, do you have any better idea? I was inspired by `application/python-pickle`.